File: src/main/java/eu/execom/fabut/graph/IsomorphicNodePair.java
package eu.execom.fabut.graph;

import eu.execom.fabut.pair.Pair;

/**
 * Class representing object pair from {@link IsomorphicGraph}.
 * 
 * @author Dusko Vesin
 * @author Nikola Olah
 * @author Bojan Babic
 * @author Nikola Trkulja
 */
public class IsomorphicNodePair extends Pair {

    /**
     * Default Isomorphic node pair constructor.
     * 
     * @param expected
     *            object
     * @param actual
     *            object
     */
    public IsomorphicNodePair(final Object expected, final Object actual) {
        super(expected, actual);
    }

    @Override
    public boolean equals(final Object arg0) {
        try {
            final IsomorphicNodePair node = (IsomorphicNodePair) arg0;
            return node.getActual() == getActual() && node.getExpected() == getExpected();
        } catch (final Exception e) {
            return false;
        }
    }

}

File: src/main/java/eu/execom/fabut/graph/IsomorphicNodePair.scala
package eu.execom.fabut.graph

import eu.execom.fabut.pair.Pair

case class IsomorphicNodePair(override val expected: AnyRef, override val actual: AnyRef) extends Pair {

  override def equals(objectInstance: Any) = {
    val that = objectInstance.asInstanceOf[IsomorphicNodePair]
    if ((this.expected eq that.expected) && (this.actual eq that.actual)) true else false
  }
}